packages/gatsby-source-contentful/src/download-with-retry.js

```diff
@@ -0,0 +1,40 @@
+const axios = require(`axios`)
+const rax = require(`retry-axios`)
+const { default: PQueue } = require(`p-queue`)
+
+const queue = new PQueue({
+  concurrency: 100,
+})
+
+export default async function downloadWithRetry(requestConfig, reporter) {
+  if (!requestConfig.url) {
+    throw new Error(`requestConfig.url is missing`)
+  }
+  const axiosInstance = axios.create()
+
+  axiosInstance.defaults.raxConfig = {
+    instance: axiosInstance,
+    onRetryAttempt: err => {
+      const cfg = rax.getConfig(err)
+      reporter.verbose(
+        `Retry attempt #${cfg.currentRetryAttempt} for ${requestConfig.url}`
+      )
+    },
+  }
+  rax.attach(axiosInstance)
+
+  try {
+    const result = await queue.add(() =>
+      axiosInstance.get(requestConfig.url, {
+        ...requestConfig,
+        // This is required as we should not set `testEnvironment: "node"`
+        // in jest.config.js just to test this properly
+        adapter: require(`axios/lib/adapters/http`),
```

Comment:
Why shouldn't we set it to node for this test suite? If not can we move it to the top?

Reply:
Because I don't know a way to set it just for this plugin. The global jest config expects a different test environment, but this test environment makes axios use whatwg-fetch which does not properly work on node